sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -190,3 +190,35 @@ def test_zero_division_hessians(data):
     X, y = data
     gb = HistGradientBoostingClassifier(learning_rate=100, max_iter=10)
     gb.fit(X, y)
+
+
+@pytest.mark.parametrize('GradientBoosting, data', [
+    (HistGradientBoostingClassifier,
+     make_classification(n_samples=10001, random_state=0, n_features=2,
+                         n_informative=2, n_redundant=0)),
+    (HistGradientBoostingRegressor,
+     make_regression(n_samples=10001, random_state=0, n_features=2,
+                     n_informative=2))]
+)
+def test_small_trainset(GradientBoosting, data):
+    # Make sure that a small trainset has the expected length (10k samples)
+    X, y = data
+    gb = GradientBoosting(random_state=42)
+    X_small, y_small = gb._get_small_trainset(X, y, seed=42)
+    assert X_small.shape[0] == 10000
+    assert y_small.shape[0] == 10000
+
+
+def test_stratification_small_trainset():
+    # Make sure that the small trainset is stratified
+    n_samples = 20000
+    class_one_prop = 0.1
+    rng = np.random.RandomState(42)
+    X = rng.randn(n_samples).reshape(n_samples, 1)
+    y = np.asarray(
```

Comment:
use np.random.binomial?

Reply:
Mainly I don't like that it's not shuffled. If you really want an exact proportion (though I doubt it's really usefull) maybe use something like

```
y = np.zeros(...)
y[:n_samples * class_one_prop] = 1
shuffle(y)
```